conbench/tests/app/test_auth.py

```diff
@@ -129,3 +137,124 @@ def test_logout_unauthenticated(self, client):
         # cannot get users page after
         response = client.get("/users/", follow_redirects=True)
         self.assert_login_page(response)
+
+
+class TestLoginOIDC(_asserts.AppEndpointTest):
+    def test_login_page_shows_sso_link(self, client):
+        r = client.get("/login/", follow_redirects=True)
+        assert "Google Login" in r.text
+
+    def test_oidc_flow_against_dex(self, client):
```

Comment:
This piece of code exercises the entire OIDC flow from the user agent's (UA) point of view.

What is the UA? For clarity, there are three parties:

- service provider/relying party (RP, conbench)
- identity provider/ OIDC provider (OP, e.g. Dex)
- user agent (UA, the HTTP client/browser that the human interacts with)